src/classic/element/ReactElementValidator.js

```diff
@@ -213,19 +199,21 @@ function validateChildKeys(node, parentType) {
   } else if (node) {
     var iteratorFn = getIteratorFn(node);
     // Entry iterators provide implicit keys.
-    if (iteratorFn && iteratorFn !== node.entries) {
-      var iterator = iteratorFn.call(node);
-      var step;
-      while (!(step = iterator.next()).done) {
-        if (ReactElement.isValidElement(step.value)) {
-          validateExplicitKey(step.value, parentType);
+    if (iteratorFn) {
```

Comment:
Kind of. Without this change, a Map object would be iterated over as if it was a keyed object.

I fixed it so that plain objects are silently ignored by detecting if they have an element child. A heuristic. So I think that this should work again. However, we might as well ignore Maps since they're likely not fragments.